text/0000-default-field-values.md

```diff
@@ -673,7 +707,35 @@ all the following rules apply when type-checking:
 
 1. The expression `def` must be a constant expression.
 
-2. The expression `def` must coerces with the type `ty`.
+2. The expression `def` must coerce to the type `ty`.
+
+3. Generic parameters of the current items are accessible
+```rust
+struct Bar<const A: usize> {
+    field: usize = A,
+}
+```
+
+4. Default const expressions are *not* evaluated at definition time, only
+   during instantiation. This means that the following will not fail to compile:
```

Comment:
I think that we can conceivably mir evaluate the `const` expression and differentiate between "couldn't be evaluated because of missing context" and "evaluated to failure", and lint in the later case. I'd be in favor of such a lint.

Reply:
Mentioned the lint